Packs/Zimperium/Integrations/Zimperium/Zimperium.py

```diff
@@ -0,0 +1,661 @@
+    verbose = str(args.get('verbose')) == 'true'
+
+    events = client.events_search_request(query, size, page, verbose)
+    events_data = events.get('content')
+
+    if not verbose:
+        for event_data in events_data:
+            event_data.pop('eventDetail', None)
+
+    table_name = ''
+    if not events.get('last'):
+        table_name = f' (To get the next events, run the command with the next page)'
+    headers = ['eventId', 'eventName', 'eventState', 'incidentSummary', 'severity', 'persistedTime']
+    readable_output = tableToMarkdown(name=f"Users{table_name}:", t=events_data, headers=headers, removeNull=True)
+
+    command_results = CommandResults(
+        outputs_prefix='Zimperium.Events',
+        outputs_key_field='eventId',
+        outputs=events_data,
+        readable_output=readable_output,
+        raw_response=events
+    )
+
+    return command_results
+
+
+def fetch_incidents(client: Client, last_run: dict, first_fetch_time: str, max_fetch: str = '50') -> Tuple[dict, list]:
+    """
+    This function will execute each interval (default is 1 minute).
+
+    Args:
+        client (Client): Zimperium client
+        last_run (dateparser.time): The greatest incident created_time we fetched from last fetch
+        first_fetch_time (dateparser.time): If last_run is None then fetch all incidents since first_fetch_time
+        max_fetch: max events to fetch
+
+    Returns:
+        next_run: This will be last_run in the next fetch-incidents
+        incidents: Incidents that will be created in Demisto
+    """
+    timestamp_format = '%Y-%m-%dT%H:%M:%S.%fZ'
+    if not last_run:  # if first time fetching
+        next_run = {
+            'time': parse_date_range(first_fetch_time, date_format=timestamp_format)[0],
+            'last_event_ids': []
+        }
+    else:
+        next_run = last_run
+
+    incidents = []
+
+    events = client.events_search_request(query=f"persistedTime=gt={next_run.get('time')}",
+                                          size=max_fetch, page='0', verbose=False)
+    events_data = events.get('content')
+
+    if events_data:
+        last_event_ids = last_run.get('last_event_ids', [])
+        for event_data in events_data:
+            event_data.pop('eventDetail', None)  # deleting eventDetail to not load the context
+            event_id = event_data.get('eventId')
+            if event_id not in last_event_ids:  # check that event was not fetched in the last fetch
+                event_created_time = parse(event_data.get('persistedTime'))
+                incident = {
+                    'name': event_data.get('incidentSummary'),
+                    'occurred': event_created_time.strftime(timestamp_format),
+                    'severity': event_severity_to_dbot_score(event_data.get('severity')),
+                    'rawJSON': json.dumps(event_data)
+                }
+                incidents.extend([incident])
+                last_event_ids.extend([event_id])
```

Comment:
no order to allow you to filter by time?

Reply:
You are right, I refactored accordingly.